MinGW_W64 CI job is failing

Relevant log snippet:
```
[ 42%] Building C object CMakeFiles/iio.dir/dns_sd_windows.c.obj
[ 44%] Building C object CMakeFiles/iio.dir/xml.c.obj
D:\a\1\s\xml.c: In function 'libiio_cleanup_xml_backend':
D:\a\1\s\xml.c:459:9: error: 'xmlMemoryDump' is deprecated [-Werror=deprecated-declarations]
  459 |         xmlMemoryDump();
      |         ^~~~~~~~~~~~~
In file included from C:/msys64/mingw64/include/libxml2/libxml/tree.h:28,
                 from C:/msys64/mingw64/include/libxml2/libxml/parser.h:16,
                 from D:\a\1\s\xml.c:12:
C:/msys64/mingw64/include/libxml2/libxml/xmlmemory.h:161:9: note: declared here
  161 |         xmlMemoryDump   (void);
      |         ^~~~~~~~~~~~~
cc1.exe: all warnings being treated as errors

```

--- a/xml.c
+++ b/xml.c
@@ -456,5 +456,7 @@ void libiio_cleanup_xml_backend(void)
 	 * detect a memory leak.
 	 */
 	xmlCleanupParser();
+#if LIBXML_VERSION < 21200  /* Version 2.12.0 */
 	xmlMemoryDump();
+#endif
 }

